### PR TITLE
Add subtitle and link to DS for va-process-list

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Chromatic

https://40111-pl--60f9b557105290003b387cd5.chromatic.com

## Description
Part of https://github.com/department-of-veterans-affairs/va.gov-team/issues/40111

## Testing done
N/A

## Screenshots
Screen Shot 2022-05-03 at 11 56 52

## Acceptance criteria
- [x] Add subtitle "Process list web component"
- [x] Add `View guidance for the Process list component in the Design System`

## Definition of done
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/storybook/stories/va-process-list.stories.jsx [hunk 1]
```diff
@@ -8,9 +8,14 @@ export default {
...
   parameters: {
     docs: {
-      /* eslint-disable-next-line react/display-name */
-      page: () => <StoryDocs data={processListDocs} />,
+      description: {
+        component:
+          `<a className="vads-c-action-link--blue" href="https://design.va.gov/components/process-list">View guidance for the Process list component in the Design System</a>` +
+          `\n` +
+          'This component expects `<li>` elements as its children.',
+      },
```
Comment (bkjohnson) [inline packages/storybook/stories/va-process-list.stories.jsx:16]: Is it possible to update `<StoryDocs>` in a way where we can still rely on the JSDoc comment in the code while also having this action link?

Comment (andresriveratoro) [reply packages/storybook/stories/va-process-list.stories.jsx:16]: We can, but I'm running into issues related to styling that I haven't been able to solve. It seems not all styles are being applied.

```diff
diff --git a/packages/storybook/stories/va-process-list.stories.jsx b/packages/storybook/stories/va-process-list.stories.jsx
index 752b5e0..0bdcdeb 100644
--- a/packages/storybook/stories/va-process-list.stories.jsx
+++ b/packages/storybook/stories/va-process-list.stories.jsx
@@ -8,13 +8,19 @@ export default {
   title: 'Components/va-process-list',
   parameters: {
     docs: {
-      description: {
-        component:
-          `<a className="vads-c-action-link--blue" href="https://design.va.gov/components/process-list">View guidance for the Process list component in the Design System</a>` +
-          `\n` +
-          'This component expects `<li>` elements as its children.',
-      },
+      page: () => (
+        <StoryDocs
+          data={{
+            ...processListDocs,
+            guidance: {
+              componentName: 'Process list',
+              url: 'https://design.va.gov/components/process-list',
+            },
+          }}
+        />
+      ),
     },
+
     componentSubtitle: `Process list web component`,
   },
 };
diff --git a/packages/storybook/stories/wc-helpers.jsx b/packages/storybook/stories/wc-helpers.jsx
index 0302df3..ec6a98b 100644
--- a/packages/storybook/stories/wc-helpers.jsx
+++ b/packages/storybook/stories/wc-helpers.jsx
@@ -100,16 +100,27 @@ export const propStructure = comp => {
   return { ...props, ...events, ...listenersWithDescriptions };
 };
 
+function Guidance({ data: guidance }) {
+  return (
+    <a className="vads-c-action-link--blue" href={guidance.url}>
+      View guidance for the {guidance.componentName} component in the Design
+      System
+    </a>
+  );
+}
+
 /**
  * Return a component with Storybook docs blocks in a standard order.
  * Accepts a JSON object as a prop representing component information
  */
 export function StoryDocs({ data }) {
   const noArgs = data.props.length == 0;
+  const guidance = data.guidance;
   return (
     <>
       <Title />
       <Subtitle />
+      {guidance && <Guidance data={guidance} />}
       <Description markdown={data.docs} />
       <Primary />
       {!noArgs && <ArgsTable story={PRIMARY_STORY} />}

```

This is how it looks:
Screen Shot 2022-05-04 at 18 06 05

This is how it should look (using Modal for example only)
Screen Shot 2022-05-04 at 18 07 03

It's not applying these dynamically generated CSS classes:
Screen Shot 2022-05-04 at 18 07 55

Comment (andresriveratoro) [reply packages/storybook/stories/va-process-list.stories.jsx:16]: Nevermind, that's the styling that we DO want after consulting Sketch. It's the opposite situation - the current action links underneath the component subtitle are being displayed incorrectly.

**Comment by andresriveratoro** [comment]
Updated `StoryDocs` to display a link to the DS if data object includes an object property named `guidance` with keys `componentHref` and `componentName`.

**Review by pdavies88 — APPROVED** [review]

File: packages/storybook/stories/wc-helpers.jsx [hunk 2]
```diff
@@ -100,19 +100,37 @@ export const propStructure = comp => {
...
+}
+
 /**
  * Return a component with Storybook docs blocks in a standard order.
  * Accepts a JSON object as a prop representing component information
  */
 export function StoryDocs({ data }) {
-  const noArgs = data.props.length == 0;
+  const args = data && data.props && data.props.length > 0;
+  const guidance = data && data.guidance;
```
Comment (bkjohnson) [inline packages/storybook/stories/wc-helpers.jsx:125]: Do we have babel set up for optional chaining in storybook?

```suggestion
  const args = data?.props?.length > 0;
  const guidance = data?.guidance;
  ```

Comment (andresriveratoro) [reply packages/storybook/stories/wc-helpers.jsx:125]: It looks like we do. 👍 
https://github.com/storybookjs/storybook/blob/next/lib/core-common/src/utils/babel.ts#L41

**Pull request merged**